lib/record_store/zone.rb

```diff
@@ -21,19 +21,74 @@ def each(&block)
 
       def find(name)
         return unless File.exists?(zone_path = "#{RecordStore.zones_path}/#{name}.yml")
-        Zone.from_yaml_definition(*YAML.load_file(zone_path).first)
+        load_yml_zone_definition(zone_path).first.last
+      end
+
+      def write(name, config:, records:, format: :file)
+        raise ArgumentError, "format must be :directory or :file" unless %i(file directory).include?(format)
+        name = name.chomp('.')
+        zone_file = "#{RecordStore.zones_path}/#{name}.yml"
+        zone = { name => { config: config.to_hash } }
+        records = records.map(&:to_hash).sort_by! {|r| [r.fetch(:fqdn), r.fetch(:type), r[:nsdname] || r[:address]]}
+
+        zone[name][:records] = records if format == :file
+        write_yml_file(zone_file, zone)
+        remove_record_files(name)
+        write_record_files(name, records) if format == :directory
       end
 
       private
 
-      def load_yaml_file(filename)
+      def write_yml_file(filename, data)
+        lines = data.deep_stringify_keys.to_yaml.lines
+        lines.shift if lines.first == "---\n"
+        File.write(filename, lines.join)
+      end
+
+      def load_yml_zone_definition(filename)
         result = {}
+        dir = File.dirname(filename)
         YAML.load_file(filename).each do |name, definition|
```

Comment:
I'll do that in a follow-up PR, along with having `find` just go through `defined`,  if that's ok.